app/javascript/packs/admin/editUser.jsx

```diff
@@ -28,21 +32,26 @@ const openModal = async (event) => {
 
   const { h, render } = preact;
 
+  if (modalContainer) {
+    // We've loaded a modal at least once in the modal container, so unmount it before rendering a new one.
+    // This allows us to only ever append one div to the body.
+    render(null, modalContainer);
+  }
+
+  const { modalTitle, modalSize, modalContentSelector } = dataset;
+
   // TODO: Where are we pulling the modal body from?
   render(
-    <Modal title={trigger.dataset.modalTitle} size={trigger.dataset.modalSize}>
+    <Modal title={modalTitle} size={modalSize}>
       <div
         // eslint-disable-next-line react/no-danger
         dangerouslySetInnerHTML={{
-          __html: document.querySelector(trigger.dataset.modalContentSelector)
-            .innerHTML,
+          __html: document.querySelector(modalContentSelector).innerHTML,
         }}
       />
     </Modal>,
     modalContainer,
   );
 };
 
-modalTriggers.forEach((trigger) =>
-  trigger.addEventListener('click', openModal),
-);
+document.body.addEventListener('click', openModal);
```

Comment:
Watching a container element for an event is preferred because only one event is added vs. n number of modal trigger events.